app/models/image.rb

```diff
@@ -5,7 +5,8 @@ class Image < ApplicationRecord
   validates :alt_text, presence: true, allow_blank: true, length: { maximum: 255 }, unless: :skip_main_validation?
   validates :image_data, presence: { message: "must be present" }
 
-  after_destroy :destroy_image_data_if_required
+  after_destroy :destroy_image_data_if_required, :update_lead_image_association
+  after_create :update_lead_image_association
```

Comment:
it feels like these are two times outside of a user choosing toggling lead image visibility that a lead_image_id may need to be set. Does that sound right?

Reply:
I think so. Out of interest, in the context of an image being created or destroyed, do we have access to an edition (i.e. is the route like `/editions/{id}/images/{id}`, with the edition available in the controller)? If we do I'd be tempted to create a method on the edition that updates the lead image association and deletes/creates the image explicitly. I'm inclined to avoid model callbacks wherever possible. In general, the `Image` class meddling with the edition state doesn't feel great to me - it feels like edition should be an [aggregate](https://martinfowler.com/bliki/DDD_Aggregate.html) of sorts and shouldn't be modified by direct access to its child associations